src/cargo/util/context/mod.rs

```diff
@@ -1174,6 +1174,9 @@ impl GlobalContext {
         let cli_target_dir = target_dir.as_ref().map(|dir| Filesystem::new(dir.clone()));
         self.target_dir = cli_target_dir;
 
+        self.shell()
+            .set_report_unicode(self.unstable_flags.rustc_unicode)?;
```

Comment:
This flag shouldn't be unilaterally enabling unicode support but making it subject to either `Shell::out_unicode` or `Shell:err_unicode`.